### PR TITLE
add login mockup

### PR DESCRIPTION
**Pull request by kaozdl**
This comit adds a login mockup where i intend to test the login functionality.
Further in the developement this code may be refactored to work propperly with react-native-navigation and redux.
Since the induction is building the app from scratch, i consider this is the way to go.

### CODE REVIEW

File: components/common/button/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {Text, View, TouchableOpacity } from 'react-native';
+import styles from './styles';
+
+const Button = ({name, action}) => (
+  <View>
```
Comment (aguscha333) [inline components/common/button/index.js]: why do you need to wrap it in a view?

File: components/common/button/styles.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+    button: {
```
Comment (aguscha333) [inline components/common/button/styles.js]: we use 2 spaces indentation

File: components/common/button/styles.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        backgroundColor: 'black',
```
Comment (aguscha333) [inline components/common/button/styles.js]: colors should be stored in a constants file and imported from there

File: components/common/input/styles.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+    input: {
+        borderColor: 'black',
```
Comment (aguscha333) [inline components/common/input/styles.js]: check for colors and indentation

File: App.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import { View } from 'react-native';
+
+
+import SignIn from './screens/signIn';
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (aguscha333) [inline App.js]: don't understand what you are doing here. In any case it doesn't seem like sth that should go in this file. Let's chat about it.

File: components/common/input/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { View, Text, TextInput } from 'react-native';
+import styles from './styles';
+
+const Input = ({ handler, placeholder, label, bindTo }) => (
+  <View style={{ padding: 10 }}>
```
Comment (aguscha333) [inline components/common/input/index.js]: are you using `redux-form`? not that you should but if you are I don't understand what you are doing here.

File: App.js (hunk 5 above)
Comment (kaozdl) [reply App.js]: im mantaining the state here but this is temporal.
The idea is to refactor this code further in the developement.
For now it's like this so i can test some stuff without too much boilerplate.

File: components/common/button/styles.js (hunk 2 above)
Comment (kaozdl) [reply components/common/button/styles.js]: Noted, will fix it

File: components/common/button/index.js (hunk 1 above)
Comment (kaozdl) [reply components/common/button/index.js]: Noted, i will fix this.

File: components/common/input/index.js (hunk 6 above)
Comment (kaozdl) [reply components/common/input/index.js]: im not using redux-form at the moment.

File: screens/signIn/index.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+      key: "NOT LOGGED IN",
+    }
+  }
+  handleInput = (inputName, text) => this.setState({ [inputName]: text });
+  handleSubmit = async () => {
+    let response = await fetch('https://kaozdl-target.herokuapp.com/api/v1/rest-auth/login/', {
```
Comment (acollazomayer) [inline screens/signIn/index.js]: This await is unnecessary since you are using then and catch to handle the promise.

**Pull request merged**